### DISCUSSION TITLE
Is anyone else getting this error on signal-cli-0.8.1?

### DISCUSSION THREAD

**Discussion by firdausa-iprogrammer** [AsamK/signal-cli#559] [General]
I just updated to signal-cli-0.8.1 today.
Getting this error -
```
Error loading or creating state file: 'long[] org.signal.client.internal.Native.IdentityKeyPair_Deserialize(byte[])' (UnsatisfiedLinkError).
```
Can't register or anything.

**Comment by AsamK** [comment]
looks like you're missing libsignal-client lib:
https://github.com/AsamK/signal-cli/wiki/Provide-native-lib-for-libsignal

There should be a more detailed error message, but looks like it's not working correctly...

**Reply by firdausa-iprogrammer** [reply]
I actually did compile the libsignal-client, but like you're suggesting it might not be working correctly.
I used the latest Java version - 0.1.7
However I also see Swift and ts, so I'm not sure which to use.

**Reply by AsamK** [reply]
Check the version of signal-client....jar in the signal-cli lib folder, it should be 0.2.3

**Reply by firdausa-iprogrammer** [reply]
Okay so I think I've got my mistake, I was trying to build the latest version of libsignal-client instead of the version matching the version that is there in the release. I.E. I built 0.3.2 instead of 0.2.3.
I really do apologize for such a stupid mistake, I missed that point in the documentation.
